hw3/exercise-3/src/components/Houses.js

```diff
@@ -0,0 +1,311 @@
+import React, { useEffect, useState } from "react";
+import Chart from "chart.js/auto";
+import { CategoryScale } from "chart.js";
+import { Doughnut } from "react-chartjs-2";
+import "./Houses.css";
+
+const backgroundColors = [
+	"rgba(54, 162, 235, 0.8)", //targaryen
+	"rgba(255, 206, 86, 0.8)", //tarly
+	"rgba(255, 99, 132, 0.8)", //stark
+	"rgba(75, 192, 192, 0.8)", //baratheon
+	"rgba(153, 102, 255, 0.8)", // lannister
+	"rgba(255, 159, 64, 0.8)", // grejoy
+	"rgba(199, 199, 199, 0.8)", // clegane
+	"rgba(83, 102, 255, 0.8)", //house belish
+	"rgba(40, 159, 64, 0.8)", //seaworth
+	"rgba(150, 150, 199, 0.8)", /// unknown
+	"rgba(78, 52, 199, 0.8)", //tyrell
+	"rgba(54, 54, 54, 0.8)", // free folk
+	"rgba(255, 206, 186, 0.8)", //tarth
+	"rgba(255, 99, 132, 0.8)", // naathi
+	"rgba(75, 275, 275, 0.8)", // bolton
+	"rgba(100, 70, 70, 0.8)", // naharis
+	"rgba(255, 159, 64, 0.8)", // lorath
+	"rgba(199, 100, 199, 0.8)", // Mormont
+	"rgba(183, 255, 255, 0.8)", // sparrow
+	"rgba(40, 259, 164, 0.8)", // viper
+	"rgba(210, 200, 100, 0.8)", //sand
+	"rgba(128, 0, 128, 0.8)", // worm
+	"rgba(255, 15, 150, 0.8)", // qyburn
+	"rgba(0, 128, 0, 0.8)", // bronn
+];
+
+const borderColors = [
+	"rgba(54, 162, 235, 0.8)", //targaryen
+	"rgba(255, 206, 86, 0.8)", //tarly
+	"rgba(255, 99, 132, 0.8)", //stark
+	"rgba(75, 192, 192, 0.8)", //baratheon
+	"rgba(153, 102, 255, 0.8)", // lannister
+	"rgba(255, 159, 64, 0.8)", // grejoy
+	"rgba(199, 199, 199, 0.8)", // clegane
+	"rgba(83, 102, 255, 0.8)", //house belish
+	"rgba(40, 159, 64, 0.8)", //seaworth
+	"rgba(150, 150, 199, 0.8)", /// unknown
+	"rgba(78, 52, 199, 0.8)", //tyrell
+	"rgba(54, 54, 54, 0.8)", // free folk
+	"rgba(255, 206, 186, 0.8)", //tarth
+	"rgba(255, 99, 132, 0.8)", // naathi
+	"rgba(75, 275, 275, 0.8)", // bolton
+	"rgba(100, 70, 70, 0.8)", // naharis
+	"rgba(255, 159, 64, 0.8)", // lorath
+	"rgba(199, 100, 199, 0.8)", // Mormont
+	"rgba(183, 255, 255, 0.8)", // sparrow
+	"rgba(40, 259, 164, 0.8)", // viper
+	"rgba(210, 200, 100, 0.8)", //sand
+	"rgba(128, 0, 128, 0.8)", // worm
+	"rgba(255, 15, 150, 0.8)", // qyburn
+	"rgba(0, 128, 0, 0.8)", // bronn
+];
+
+const Houses = () => {
+	const [house, setHouse] = useState({ labels: [], values: [] });
+
+	const url = "https://thronesapi.com/api/v2/Characters";
+	const fetchData = () => {
+		fetch(url)
+			.then((response) => {
+				return response.json();
+			})
+			.then((data) => {
+				mapToHouses(data);
+			});
+	};
+
+	const mapToHouses=(data) => {
+		
+				const family = data.map((item) => item.family);
+				let hashMap = new Map();
```

Comment:
Is it possible to extract the hashmap function to outside of the component to improve readability?